### PR TITLE
DOP-3791

### PR DESCRIPTION
**Pull request by bianca-laube**
### Stories/Links:

DOP-3791

### Current Behavior:

https://www.mongodb.com/docs/search/?q=%24and

### Staging Links:
with feature flag ON:
https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=%24and
(with feature flag OFF then it should look the same as prod)

#### Changes
- Now when you have a search term and go to add a filter it will load the loading skeleton while its searches for the new filtered search result list
- it should load the skeleton for when you add a filter category and for when you change the category version
- note: I didnt actually change the following files for this ticket: (I merged DOP-3791 with master so it updated these extra files so thats why it looks like they are changed when comparing to DOP-3786) 
         - `package.json`
         - `package-lock.json`
         - `DeprecatedVersionSelector.js`
         - `get-site-url.js`
         - `DeprecatedVersionSelector.test.js`
         - `get-site-url.test.js`

#### Testing
The following list is something I have been using to keep track of testing the website. can be used as a guide for when reviewing. 
1. Empty search on top nav search loads empty page
     - https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html
     - https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=
2. Search of "failed%20to%20find%20a%20plan"* loads to a no results found modal
     - https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q="failed%20to%20find%20a%20plan"*
3. Search from the top nav bar shell  (or right into the url) to see if it loads properly
     - https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/bianca.laube/DOP-3791/search/index.html?q=shell
4. Make a new search from the persistent search bar of with the input `$and` make sure it loads properly
5. Add a filter make sure skeleton shows
6. change the filter version and make sure the the skeleton loads again
7. Add a filter that shows no results (for example: the input `$and` with the filter `atlas data lake`
8. Do an empty search from the persistent search bar and make sure it shows the no results modal

### CODE REVIEW

**Review by mmeigs — APPROVED** [review]
LGTM!

**Review by rayangler — COMMENTED** [review]
One nitpick for now, but feel free to ignore if it doesn't feel worth the effort.

I do have one question about the loading state and the design. I'm not sure if this is a blocker based on the A/C of the ticket:

It seems like when switching between filters after a search, the loading state is also applied to the filters. Based on the designs, it seems like only the results should be in the loading state while the filters are already loaded in. Is this intentional?

File: src/components/SearchResults/SearchResults.js [hunk 1]
```diff
@@ -299,10 +299,16 @@ const SearchResults = () => {
     setSearchFilter(searchProperty);
   }, [search, firstRenderComplete]);
 
+  // add loading skeleton when new filter selected and loading results
+  useEffect(() => {
+    newSearchInput && setSearchFinished(false);
```
[the remark this answers is not in the document]
Comment (bianca-laube) [reply src/components/SearchResults/SearchResults.js]: Hey raymund,

Thanks for catching that! I fixed the loading skeleton so it doesnt changer the filter section anymore when the feature flag is on (on prod it still loads the filter section). and i updated this to be an if statement. :)

**Review by rayangler — APPROVED** [review]
LGTM! Thanks for fixing the bug with the duplicate search filter components! Tested locally and the filters no longer have flashing options

**Pull request merged**